enterprise/server/workflow/service/service.go

```diff
@@ -641,7 +641,7 @@ func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tabl
 	os := strings.ToLower(workflowAction.OS)
 	computeUnits := *workflowsMacComputeUnits
 	// Use the CI runner image if the OS supports containerized actions.
-	if os == "" || os == platform.LinuxOperatingSystemName {
+	if !workflowAction.SelfHosted && (os == "" || os == platform.LinuxOperatingSystemName) {
```

Comment:
I think maybe this check should move to the `if *enableFirecracker {` check below? (we still want to apply the configured linux compute units if the OS is linux)

Reply:
Good call, done.